ArduPilot/ardupilot
# AP_DDS: Status Message

- Issue: https://github.com/ArduPilot/ardupilot/issues/28301
- Creates a new `/ap/status` message
- Publishes the topic only when the status changes.
- The status reports a flag `external_control`, which will be filled by or after https://github.com/ArduPilot/ardupilot/pull/28429.

**Test**
- Launch the SITL.
- Open a new terminal and echo the new topic.
- Verify that only one message is published.
- Now change the vehicle's mode and verify that a new message has been published.
- Arm and takeoff, verify that a new message has been published only when `armed` flag changes.

Test a failsafe:
- Change the parameter `BATT_LOW_VOLT` to 11.0
- Set the parameter `SIM_BATT_VOLTAGE` to 10.6.
- Verify that the status topics reports a failsafe of type 22
- Disable the RC setting `SIM_RC_FAIL` to 1.
- Verify that the failsafe array now reports 22 and 21

![image](https://github.com/user-attachments/assets/2a777baf-8104-473b-bad6-f6d3db8d0723)
